### ISSUE TITLE
Don't default Hindi if country is detected as India

### ISSUE THREAD

**Issue by eggjoystick** [calamares/calamares#1284]
**Is your feature request related to a problem? Please describe.**
Paraphrasing [a Reddit user from India](https://old.reddit.com/r/ManjaroLinux/comments/e64bpv/a_sincere_request_from_india/):
If the location is detected as India, the interface auto switches to Hindi. One of the most common mistakes people make about India is assuming there is a common language or a national language. In fact, there are over 20 major languages and literally hundreds of different dialects. Yes, Hindi is one of the most widely spoken languages in India but only in the north. Note "one of the". Hindi is not at all common in southern or eastern parts of India. Many Indians can only read/write in English but can speak multiple other languages.

**Describe the solution you'd like**
Instead of defaulting to Hindi, please change it to at least English (India). There is a greater chance of not making the interface difficult to use.

**Describe alternatives you've considered**
- Original request was made on Reddit by user kookykau.
- I posted about the issue on the Manjaro Linux forum and was directed here as it is a Calameres issue rather than a distribution-specific issue.

**Additional context**
https://forum.manjaro.org/t/a-sincere-language-request-from-india/113970
https://old.reddit.com/r/ManjaroLinux/comments/e64bpv/a_sincere_request_from_india/

**Comment by adriaandegroot** [comment]
This is a perfectly reasonable request. I'll task **you**, @eggjoystick , with communicating downstream (e.g. on Reddit and/or the Manjaro forums, because I don't go there).

A little background: the welcome module is usually the first one that Calamares runs (it says hi, does some background checking, and offers the language-selection drop-down). It can be configured to use GeoIP, and I'm pretty sure Manjaro does so (@philmmanjaro can say for sure). If GeoIP is used, then Calamares uses the [Unicode CLDR](http://www.unicode.org/cldr/charts/latest/supplemental/likely_subtags.html) data to look up, based on the country, what the most likely language is. The CLDR data says *India: Hindi*.

So that's how we get to this situation. And of course it's bogus, which my friends from Kerala State could also point out (those friends are the reason Malayalam is in the translations list).

The language-change icon is an international standard so it's supposed to be easy to find the change-language-here button; it's still inconvenient if the language is wrong for you. Belgium, by the way, has a similar problem: Unicode says Dutch, which the 35% of Walloons don't appreciate.

So, how to fix:
 - (*simple*) Change the extracted data table, `src/libcalamares/locale/CountryData_p.cpp` and substitute in English. There's no provision for more subtlety in language selection than that.
 - (*medium*) Adjust the table to also have *none* languages; if your GeoIP returns that country, just don't change the language from the LANG or whatever setting that is already there. This just means that Calamares will match the language of its environment.
 - (*complicated*) Chase GeoIP providers and see if they provide more detailed information and then use country + state to do a more accurate guess. It's still going to be a guess, though.

I like the medium solution (it should also be easy to patch back if you happen to have a distribution aimed at a Hindi-speaking area).

**Comment by eggjoystick** [comment]
Hello, and thank you for your reply. Also appreciate the background information regarding Unicode CLDR as this was all new to me.

I've already updated the Reddit and Manjaro forum topics regarding having opened this issue here.

**Comment by adriaandegroot** [comment]
@eggjoystick I also spoke with @bhush9 who re-iterated that Hindi as default only makes things **more** difficult. I've chosen the *medium* approach.

.. except, thinking about it, I might have solved the problem in the wrong place. Needs a little testing here. In any case, it'll be done in 3.2.18 and should be testable from master by tomorrow morning.

**Comment by adriaandegroot** [comment]
After faking a response of "IN" in my GeoIP lookup, I can confirm that before that commit, with GeoIP switched on, Hindi is selected automatically; after that commit, with GeoIP switched on, the language does not change (e.g. if you set `LANG=ml` i the environment, it will stick with Malayalam).

**Comment by eggjoystick** [comment]
@adriaandegroot thank you! I have updated my post in the Manjaro forum and also included @philmmanjaro

**Comment by kaushik-evani** [comment]
Hi @all, I was the one who posted on reddit about this. It's wonderful that this issue was picked up so quickly and is also on its way out in the next release. To be honest, when I posted this, I was only ranting and didn't expect any actual action to be taken on it. My respect to the community has only increased.
I just want to add another point though.
As far as I have seen, the general language in computers and phones in India is English. Except in word processors and phone keypads, I have not seen the use of native languages at all. The UI language is mostly English (except maybe in ATM machines) unlike China or Japan.
There might be some people in rural areas (or areas with lower English literacy, and even among those, I'm willing to bet it's only older people) who may have configured their complete phones to their local language. But if you throw in a computer, and Linux, and Manjaro into the equation, I think the numbers are statistically insignificant. English is actually one of the important languages in India, what with the couple hundred years of British rule.
This of course is an educated guess based on my observations. If anyone has numbers to prove me otherwise, you are most welcome.

Cheers!

**Comment by subins2000** [comment]
Hi Adrian, I downloaded the latest KDE Neon Unstable image (`neon-unstable-20210815-0328.iso`, Calamares 3.2.42), tried it and I found this issue. The default was set to Hindi Inscript keyboard layout. I see the locale is set to [`AnyLanguage`](https://github.com/calamares/calamares/blob/calamares/src/libcalamares/locale/CountryData_p.cpp#L138) and from the previous comments the environment variable is respected to set the default layout.

I think it should pick English layout for default than Hindi Inscript layout.

![telegram-cloud-photo-size-5-6132017971091058077-y](https://user-images.githubusercontent.com/2305675/130435221-58ac6028-143b-4d7b-8c0d-fc143638377a.jpg)
